_episodes/requirements.md

```diff
@@ -18,4 +18,91 @@ By the end of this episode,
 learners should be able to
 __describe all the requirements for running a tool__
 
+
+> ## Exercise 1:
+>
+> For each of these variables, is it required in a CommandLineTool?
+>
+> A. inputs
+> B. outputs
+> C. arguments
+> D. baseCommand
+> E. requirements
+> F. hints
+> > ## Solution
+> > Just like a normal shell script, you need inputs, outputs, and a command to run. So A, B, C are definitely needed. D is sometimes needed, this is where you put the binary invocation of the tool. E and F are necessary for adding environment help, such as machine and docker requirements.
+> {: .solution}
+{: .challenge}
+
+> ## Exercise 2:
+> 
+> When working in a cloud environment, you need to specify what machine type you would like to run on. Which means the job has to have specific RAM, Cores and Disk space parameters.
+>
+> What ResourceRequirements can you specify in a CommandLineTool to satisfy the machine requirements?
+>
+> > ## Solution:
+> >
+> > coresMin and coresMax
+> > ramMin and ramMax
+> > tmpdirMin and tmpdirMax
+> > outdirMin andoutdirMax
+> {: .solution}
+{: .challenge}
```

Comment:
Lets invent a bioinformatic tool who's README specifies the machine requirements (but not precisely how CWL needs them) and ask the user questions about that